### PR TITLE
Release/state comp

### PR DESCRIPTION
**Pull request by nobody5050**

### CODE REVIEW

File: src/main/cpp/utils/TargetTracker.cpp [hunk 1]
```diff
@@ -104,7 +171,9 @@ units::inch_t TargetTracker::GetDistanceToTarget(const DetectedObject& target) {
   frc::SmartDashboard::PutNumber("TargetTracker Pixel Width", pixelWidth);
 
   auto otherDistance =
-      pixelWidth > 0
+      // TODO: Replace with constant
+      // This ensures we don't get a wildly small width (= very far away) if bounding box is wrong/missing
+      pixelWidth > 3
```
Comment (having11) [inline src/main/cpp/utils/TargetTracker.cpp:176]: Still want a constant here?

File: src/main/cpp/utils/TurnToPose.cpp [hunk 2]
```diff
@@ -54,6 +63,26 @@ units::degree_t TurnToPose::GetAngleFromOtherPose(
 void TurnToPose::SetTargetPose(frc::Pose2d pose) {
   m_startPose = m_poseGetter();
   m_targetPose = pose;
+  m_targetAngle = std::nullopt;
+}
+
+void TurnToPose::SetTargetAngleRelative(units::degree_t angle) {
+  m_startPose = m_poseGetter();
+  // auto wpiTransformation = frc::Transform2d(0_m, 0_m, angle);
```
Comment (having11) [inline src/main/cpp/utils/TurnToPose.cpp:71]: Can be deleted

File: src/main/include/Constants.h
Comment (having11) [inline src/main/include/Constants.h:1]: woohoo! It's gone

File: src/main/include/RobotContainer.h [hunk 3]
```diff
@@ -106,27 +123,34 @@ class RobotContainer {
...
+       // Sim gamepiece pose
+       VisionConstants::kSimGamepiecePose,
+       // Gamepiece rotation
+       VisionConstants::kGamepieceRotation,
+       // Trig-based distance percentage
+       VisionConstants::kTrigDistancePercentage,
+       // Area percentage threshold
+       VisionConstants::kAreaPercentageThreshold,
+       // Max # of tracked objects
+       10,
```
Comment (having11) [inline src/main/include/RobotContainer.h:148]: constant?

File: src/main/include/commands/FlywheelRampCommand.h [hunk 4]
```diff
@@ -28,7 +28,7 @@ class FlywheelRamp : public frc2::CommandHelper<frc2::Command, FlywheelRamp> {
   }
 
   void Execute() override {
-    // isFinished = m_scoring->GetMotorAtScoringSpeed(m_direction());
+    //isFinished = m_scoring->GetMotorAtScoringSpeed(m_direction());
```
Comment (having11) [inline src/main/include/commands/FlywheelRampCommand.h:31]: remove?

File: src/main/include/constants/ScoringConstants.h [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+// These should match
+constexpr double kSpeakerLowerSpeed = 1;
+constexpr double kSpeakerUpperSpeed = kSpeakerLowerSpeed;
+
+// These should also match
+constexpr double kSubwooferLowerSpeed = -0.95;
+constexpr double kSubwooferUpperSpeed = kSubwooferLowerSpeed;
+
+// nice :flushed:
+constexpr double kFeedLowerSpeed = 0.69;
```
Comment (having11) [inline src/main/include/constants/ScoringConstants.h:34]: nice

File: src/main/include/subsystems/singleaxis/BaseSingleAxisSubsystem.h [hunk 6]
```diff
@@ -44,6 +44,10 @@ class BaseSingleAxisSubsystem
 
  protected:
   bool IsMovementAllowed(double speed, bool ignoreEncoder = false) {
+    if (m_config.ignoreLimit()) return true;
+
+    ConsoleVerbose("BaseSingleAxis", "Did not ignore limit%s", "");
```
Comment (having11) [inline src/main/include/subsystems/singleaxis/BaseSingleAxisSubsystem.h:49]: Can be removed

File: src/main/include/utils/AutoChooser.h
Comment (having11) [inline src/main/include/utils/AutoChooser.h:1]: Hawt

File: src/main/include/utils/PidMotorController.h [hunk 7]
```diff
@@ -95,7 +96,7 @@ class PidMotorController {
       return;
     }
     auto rpm = units::revolutions_per_minute_t(m_maxRpm) * percentage;
-
+    ConsoleWriter.logVerbose("pidmotor", "%s: %f", m_name.c_str(), percentage);
```
Comment (having11) [inline src/main/include/utils/PidMotorController.h:99]: Remove

**Review by having11 — CHANGES_REQUESTED** [review]
Pretty good, just some minor stuff about constants mostly

**Comment by having11** [comment]
Should we also fix the Mechanism2d offsets in this PR?

File: src/main/cpp/utils/TargetTracker.cpp (hunk 1 above)
Comment (5690Programmers) [reply src/main/cpp/utils/TargetTracker.cpp:176]: naw

File: src/main/include/utils/AutoChooser.h
Comment (5690Programmers) [reply src/main/include/utils/AutoChooser.h:1]: 🥵

**Pull request merged**